test/absinthe/phase/document/complexity_test.exs

```diff
@@ -233,8 +233,8 @@ defmodule Absinthe.Phase.Document.ComplexityTest do
       errors = result.execution.validation_errors |> Enum.map(& &1.message)
 
       assert errors == [
-               "Field fooComplexity is too complex: complexity is 6 and maximum is 5",
-               "Operation ComplexityError is too complex: complexity is 6 and maximum is 5"
+               "Field fooComplexity is too complex: you asked for 6 fields and the maximum is 5",
```

Comment:
Is there an easy way to provide a single error message instead of it being scoped to a field? 

Things on my mind

1. No real need to mention complexity at all IMO, it won't mean anything to most people
2. A blanket error message of "You requested $X fields but may only request a maximum of $Y fields in a single query" would be nice. At this stage I don't think the field that pushed it over the max matters, so I'm wondering if we could ignore the node altogether. I could be misunderstanding how this works though.

Reply:
I'm a little confused how this works as well...

It's iterating over the fields and then comparing each's complexity to the `max_complexity`. I think it must sum up the complexities and update them in order during the analysis phase: https://github.com/binaryseed/absinthe/blob/custom-v1.4/lib/absinthe/phase/document/complexity/analysis.ex

Thus, it errors once a particular field goes over the complexity. Maybe the error message should be something like:

"You have hit the complexity cap at #{field}. The max number of fields you can query for is #{max} and this is the #{complexity} field."

That's still confusing, so yeah, maybe perhaps we can figure out a way to add a blanket error.